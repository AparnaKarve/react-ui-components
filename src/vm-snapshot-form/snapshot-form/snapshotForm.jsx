import React from 'react';
import { Form, Field } from 'react-final-form';
import { Form as PfForm, Button, Col, Row, ButtonGroup } from 'patternfly-react';
import PropTypes from 'prop-types';
import { required } from 'redux-form-validators';

import { FinalFormField, FinalFormTextArea, FinalFormCheckBox } from '../../forms/';

const renderTextField = (input, meta, label) => (
  <FinalFormField input={input} meta={meta} label={label} />
);

const renderTextArea = (input, meta, label) => (
  <FinalFormTextArea input={input} meta={meta} label={label} />
);

const renderCheckBox = (input, meta, label) => (
  <FinalFormCheckBox input={input} meta={meta} label={label} />
);

export const VmSnapshotForm = ({
  onSubmit,
  errorMessages,
  hideName,
  onCancel,
  nameRequired,
  descriptionRequired,
  labels,
}) => (
  <Form
    onSubmit={onSubmit}
    render={({ invalid, values, handleSubmit }) => (
      <PfForm horizontal>
        <Row>
          <Col xs={12}>
            {!hideName &&
              <Field
                validate={nameRequired && required({ msg: errorMessages.name })}
                name="name"
                id="name"
                render={({ input, meta }) => renderTextField(input, meta, labels.name)}
              />
            }
            <Field
              name="description"
              id="description"
              validate={descriptionRequired && required({ msg: errorMessages.description })}
              render={({ input, meta }) => renderTextArea(input, meta, labels.description)}
            />
            <Field
              name="snap_memory"
              id="snap_memory"
              type="checkbox"
              render={({ input, meta }) => renderCheckBox(input, meta, labels.snapMemory)}
            />
            <hr />
          </Col>
          <Col xs={12}>
            <ButtonGroup className="pull-right">
              <Button id="snap-form-submit" onClick={() => handleSubmit(values)} disabled={invalid} bsStyle="primary">{labels.create}</Button>
              <Button id="snap-form-cancel" onClick={onCancel}>{labels.cancel}</Button>
            </ButtonGroup>
          </Col>
        </Row>
      </PfForm>
    )}
  />
);

VmSnapshotForm.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  onCancel: PropTypes.func.isRequired,
  errorMessages: PropTypes.shape({
    name: PropTypes.string,
    description: PropTypes.string,
  }),
  labels: PropTypes.shape({
    name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    snapMemory: PropTypes.string.isRequired,
    create: PropTypes.string.isRequired,
    cancel: PropTypes.string.isRequired,
  }).isRequired,
  hideName: PropTypes.bool,
  nameRequired: PropTypes.bool,
  descriptionRequired: PropTypes.bool,
};

VmSnapshotForm.defaultProps = {
  hideName: false,
  nameRequired: false,
  descriptionRequired: false,
};
