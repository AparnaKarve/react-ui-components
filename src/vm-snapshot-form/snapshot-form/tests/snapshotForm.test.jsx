import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { SnapshotForm } from '../snapshotForm';

describe('Final form input component', () => {
  const initialProps = {};
  beforeEach(() => {
    initialProps.onSubmit = jest.fn();
    initialProps.onCancel = jest.fn();
    initialProps.labels = {
      name: 'Input name',
      description: 'Input description',
      snapMemory: 'Snap memory',
      create: 'Create',
      cancel: 'Cancel',
    };
    initialProps.errorMessages = {
      name: 'Name required',
      description: 'Description required',
    };
  });

  it('Should render correctly', () => {
    const tree = renderer.create(<SnapshotForm {...initialProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render with name required', () => {
    const tree = renderer.create(<SnapshotForm {...initialProps} nameRequired />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render with description required', () => {
    const tree = renderer.create(<SnapshotForm {...initialProps} descriptionRequired />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Submit button should be disabled', () => {
    const onSubmit = jest.fn();
    const wrapper = mount(<SnapshotForm {...initialProps} nameRequired onSubmit={onSubmit} />);
    wrapper.find('button#snap-form-submit').simulate('click');
    expect(onSubmit).not.toHaveBeenCalled();
  });

  it('Should call onCancel function', () => {
    const onCancel = jest.fn();
    const wrapper = mount(<SnapshotForm {...initialProps} onCancel={onCancel} />);
    wrapper.find('button#snap-form-cancel').simulate('click');
    expect(onCancel).toHaveBeenCalled();
  });

  it('Submit should return correct values', () => {
    const onSubmit = jest.fn();
    const wrapper = mount(<SnapshotForm {...initialProps} onSubmit={onSubmit} />);
    wrapper.find('button#snap-form-submit').simulate('click');
    expect(onSubmit).toHaveBeenCalled();
  });
});
