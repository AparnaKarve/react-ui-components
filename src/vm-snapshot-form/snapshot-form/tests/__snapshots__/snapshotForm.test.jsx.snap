// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Final form input component Should render correctly 1`] = `
<form
  className="form-horizontal"
>
  <div
    className="row"
  >
    <div
      className="col-xs-12"
    >
      <div
        className="form-group"
      >
        <label
          className="control-label col-xs-2"
        >
          Input name
        </label>
        <div
          className="col-xs-8"
        >
          <input
            className="form-control"
            id={undefined}
            name="name"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          className="control-label col-xs-2"
        >
          Input description
        </label>
        <div
          className="col-xs-8"
        >
          <textarea
            className="form-control"
            id={undefined}
            name="description"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder=""
            type={undefined}
            value=""
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          className="control-label col-xs-2"
        >
          Snap memory
        </label>
        <div
          className="col-xs-8"
        >
          <div
            className="checkbox"
            style={undefined}
          >
            <label
              title=""
            >
              <input
                checked={false}
                disabled={false}
                name="snap_memory"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="checkbox"
                value=""
              />
            </label>
          </div>
        </div>
      </div>
      <hr />
    </div>
    <div
      className="col-xs-12"
    >
      <div
        className="pull-right btn-group"
      >
        <button
          className="btn btn-primary"
          disabled={false}
          id="snap-form-submit"
          onClick={[Function]}
          type="button"
        >
          Create
        </button>
        <button
          className="btn btn-default"
          disabled={false}
          id="snap-form-cancel"
          onClick={[MockFunction]}
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</form>
`;

exports[`Final form input component Should render with description required 1`] = `
<form
  className="form-horizontal"
>
  <div
    className="row"
  >
    <div
      className="col-xs-12"
    >
      <div
        className="form-group"
      >
        <label
          className="control-label col-xs-2"
        >
          Input name
        </label>
        <div
          className="col-xs-8"
        >
          <input
            className="form-control"
            id={undefined}
            name="name"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="form-group has-error"
      >
        <label
          className="control-label col-xs-2"
        >
          Input description
        </label>
        <div
          className="col-xs-8"
        >
          <textarea
            className="form-control"
            id={undefined}
            name="description"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder=""
            type={undefined}
            value=""
          />
          <span
            className="help-block"
          >
            Description required
          </span>
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          className="control-label col-xs-2"
        >
          Snap memory
        </label>
        <div
          className="col-xs-8"
        >
          <div
            className="checkbox"
            style={undefined}
          >
            <label
              title=""
            >
              <input
                checked={false}
                disabled={false}
                name="snap_memory"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="checkbox"
                value=""
              />
            </label>
          </div>
        </div>
      </div>
      <hr />
    </div>
    <div
      className="col-xs-12"
    >
      <div
        className="pull-right btn-group"
      >
        <button
          className="btn btn-primary"
          disabled={true}
          id="snap-form-submit"
          onClick={[Function]}
          type="button"
        >
          Create
        </button>
        <button
          className="btn btn-default"
          disabled={false}
          id="snap-form-cancel"
          onClick={[MockFunction]}
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</form>
`;

exports[`Final form input component Should render with name required 1`] = `
<form
  className="form-horizontal"
>
  <div
    className="row"
  >
    <div
      className="col-xs-12"
    >
      <div
        className="form-group has-error"
      >
        <label
          className="control-label col-xs-2"
        >
          Input name
        </label>
        <div
          className="col-xs-8"
        >
          <input
            className="form-control"
            id={undefined}
            name="name"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder=""
            type="text"
            value=""
          />
          <span
            className="help-block"
          >
            Name required
          </span>
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          className="control-label col-xs-2"
        >
          Input description
        </label>
        <div
          className="col-xs-8"
        >
          <textarea
            className="form-control"
            id={undefined}
            name="description"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder=""
            type={undefined}
            value=""
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          className="control-label col-xs-2"
        >
          Snap memory
        </label>
        <div
          className="col-xs-8"
        >
          <div
            className="checkbox"
            style={undefined}
          >
            <label
              title=""
            >
              <input
                checked={false}
                disabled={false}
                name="snap_memory"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="checkbox"
                value=""
              />
            </label>
          </div>
        </div>
      </div>
      <hr />
    </div>
    <div
      className="col-xs-12"
    >
      <div
        className="pull-right btn-group"
      >
        <button
          className="btn btn-primary"
          disabled={true}
          id="snap-form-submit"
          onClick={[Function]}
          type="button"
        >
          Create
        </button>
        <button
          className="btn btn-default"
          disabled={false}
          id="snap-form-cancel"
          onClick={[MockFunction]}
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</form>
`;
